src/flir_adk_ethernet/EthernetCamera.cpp

```diff
@@ -123,13 +142,53 @@ bool EthernetCamera::camTypeMatches(string camType, INodeMap& nodeMapTLDevice) {
     return false;
 }
 
+bool EthernetCamera::setPTP() {
+    try {
+        INodeMap &nodeMap = _pCam->GetNodeMap();
+        CBooleanPtr ptpEnable = nodeMap.GetNode("GevIEEE1588");
+        CEnumerationPtr ptpMode = nodeMap.GetNode("GevIEEE1588Mode");
+        CEnumerationPtr ptpStatus = nodeMap.GetNode("GevIEEE1588Status");
+
+        ptpEnable->SetValue(true);
+        ptpMode->SetIntValue(Spinnaker::GevIEEE1588Mode_SlaveOnly);
+        _ptpEnabled = true;
+        
+        ROS_INFO("flir_adk_ethernet - INFO : PTP enabled flag: %d", ptpEnable->GetValue());
```

Comment:
iirc, the format codes here are wrong, as the int types for the node map are long (the enable flag is probably ok).
do you get warnings when building?
if so, fix those by fixing the %d -> %ld, etc., as appropriate.